src/SFML/Graphics/SpriteBatch.cpp

```diff
@@ -0,0 +1,267 @@
+//
+// 1. The origin of this software must not be misrepresented;
+//    you must not claim that you wrote the original software.
+//    If you use this software in a product, an acknowledgment
+//    in the product documentation would be appreciated but is not required.
+//
+// 2. Altered source versions must be plainly marked as such,
+//    and must not be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source distribution.
+//
+////////////////////////////////////////////////////////////
+
+////////////////////////////////////////////////////////////
+// Headers
+////////////////////////////////////////////////////////////
+#include <SFML/Graphics/Font.hpp>
+#include <SFML/Graphics/RenderStates.hpp>
+#include <SFML/Graphics/RenderTarget.hpp>
+#include <SFML/Graphics/SpriteBatch.hpp>
+#include <SFML/System/Err.hpp>
+
+#include <algorithm>
+
+namespace sf
+{
+////////////////////////////////////////////////////////////
+void SpriteBatch::setBatchMode(BatchMode batchMode)
+{
+    if (m_triangles.size() > 0)
+    {
+        err() << "You must clear the batch before changing the batch mode.";
+        return;
+    }
+
+    m_batchMode = batchMode;
+}
+
+
+////////////////////////////////////////////////////////////
+void SpriteBatch::batch(const Batchable& drawable, float depth)
+{
+    drawable.batch(*this, depth);
+}
+
+
+////////////////////////////////////////////////////////////
+void SpriteBatch::batch(const Vertex*    vertices,
+                        std::size_t      count,
+                        PrimitiveType    type,
+                        const Texture*   texture,
+                        const Transform& transform,
+                        float            depth)
+{
+    const bool unsupportedType = type != PrimitiveType::TriangleFan && type != PrimitiveType::TriangleStrip &&
+                                 type != PrimitiveType::Triangles;
+
+    if (unsupportedType)
+    {
+        err() << "SpriteBatch supports only triangle-based primitive types.";
+        return;
+    }
+
+    if (count == 0)
+        return;
+
+    // Split into triangles
+    switch (type)
+    {
+        case PrimitiveType::TriangleStrip:
+            for (std::size_t i = 2; i < count; i++)
+            {
+                pushTriangle(vertices[i - 2], vertices[i - 1], vertices[i], transform, texture, depth);
+            }
+            break;
+
+        case PrimitiveType::TriangleFan:
+            for (std::size_t i = 2; i < count; i++)
+            {
+                pushTriangle(vertices[0], vertices[i - 1], vertices[i], transform, texture, depth);
+            }
+            break;
+
+        case PrimitiveType::Triangles:
+            for (std::size_t i = 2; i < count; i += 3)
+            {
+                pushTriangle(vertices[i - 2], vertices[i - 1], vertices[i], transform, texture, depth);
+            }
+            break;
+
+        default:
+            // Shouldn't ever be hit
```

Comment:
Instead of this comment, you can simply omit this default case. If you add a new enum, the compiler will warn that there is an unhandled enum value.

Reply:
Omitting it would mean I'm not handling the line and point primitives. Maybe I should swap the default case with three cases for those enums, to keep the warns for future new enums?

Reply:
My bad. I assumed all cases were being handled. If you're truly confident that the default case will never get hit, add an `assert` in there to catch any instances where it might accidentally get called.

Reply:
Do I just put `assert(false)` in the default case, then?

Reply:
Yeah that should be all you need. You ought to add a message too like this:
```cpp
assert(("This is what happened", false));
```
It (ab)uses the comma operator to make sure you get some useful output when that assert fires. 
```
example.cpp:5: int main(): Assertion `("This is what happened", false)' failed.
```

Reply:
CI is complaining about the message, it seems.
```
/home/runner/work/SFML/SFML/src/SFML/Graphics/SpriteBatch.cpp:106:21: error: left operand of comma operator has no effect [-Werror=unused-value]
  106 |             assert(("A non-triangle primitive type was encountered when decomposing into triangles.", false));
```

Reply:
Oh, that's too bad. I guess we'll just have to make do with no message.

Reply:
I fixed it by casting the string to `void`. Although now that I think of it, I used a C style cast to do that.